.travis.yml

```diff
@@ -108,13 +96,55 @@ jobs:
        - chmod +x ./trillian/integration/integration_test.sh
        - ./trillian/integration/integration_test.sh
        - popd
+    - name: "presubmit"
+      script:
+       - ./scripts/presubmit.sh --no-linters --no-generate
+      env:
+       - GOFLAGS='-race'
+    - name: "presubmit (batched_queue)"
+      script:
+       - ./scripts/presubmit.sh --no-linters --no-generate
+      env:
+       - GOFLAGS='-race --tags=batched_queue'
+    - name: "presubmit (pkcs11)"
+      script:
+       - ./scripts/presubmit.sh --no-linters --no-generate
+      env:
+       - GOFLAGS='-race --tags=pkcs11'
+    - name: "integration"
+      script:
+       - ./integration/integration_test.sh && HAMMER_OPTS="--operations=50" ./integration/maphammer.sh 3
+      env:
+       - GOFLAGS='-race'
+    - name: "integration (etcd)"
+      install:
+       - go install go.etcd.io/etcd go.etcd.io/etcd/etcdctl
+      script:
+       - ./integration/integration_test.sh && HAMMER_OPTS="--operations=50" ./integration/maphammer.sh 3
+      env:
+       - GOFLAGS='-race' ETCD_DIR="${GOPATH}/bin"
+    - name: "integration (batched_queue)"
+      script:
+       - ./integration/integration_test.sh && HAMMER_OPTS="--operations=50" ./integration/maphammer.sh 3
+      env:
+       - GOFLAGS='-race --tags=batched_queue'
+    - name: "integration (pkcs11)"
+      script:
+       - ./integration/integration_test.sh && HAMMER_OPTS="--operations=50" ./integration/maphammer.sh 3
+      env:
+       - GOFLAGS='-race --tags=pkcs11' INSTALL_SOFTHSM=true
 
 services:
   - docker
   - postgresql
   - mysql
 
 before_install:
+  - |
+    if [[ "${INSTALL_SOFTHSM}" == "true" ]]; then
```

Comment:
Could this block be moved into a `before_install` inside "integration (pkcs11)" ?

Reply:
It could, but then I'd have to copy the rest of before_install also into there because Travis doesn't allow you to define before_install = ${before_install} + some_steps. You have to use the common step in its entirety, or replace it with something custom. Unless I'm missing something?

If I'm not, this way seems less onerous than duplicating the MySQL RAM disk stuff.

Reply:
Fair enough. Is there another step that you could override -- say `install`?